packages/select/src/createRemoveOnDeleteForwardPlugin.ts

```diff
@@ -0,0 +1,22 @@
+import { createPluginFactory, QueryNodeOptions } from '@udecode/plate-common';
+
+import { withCreateRemoveOnDeleteForward } from './withCreateRemoveOnDeleteForward';
+
+export type RemoveOnDeleteForwardPlugin = {
+  query?: QueryNodeOptions;
+  removeNodeIfEmpty?: boolean;
```

Comment:
```suggestion
  removeNodeWhenEmpty?: {
    query?: QueryNodeOptions;
  } | boolean
```

Reply:
@12joan let me know if you have a better suggestion

Reply:
I think `removeNodeIfEmpty` is a holdover from the select on backspace plugin (which we're keeping separate for now and which @archie9211 used as a starting point).
```suggestion
  query?: QueryNodeOptions;
```